### ISSUE TITLE
Expanding PVC value causes upgrade to fail

### ISSUE THREAD

**Issue by igorrudyk** [bitnami/charts#9671]
### Name and Version

bitnami/mongodb

### What is the problem this feature will solve?

1) `helm install mongo bitnami/mongodb -f values.yml`
2) update pvc size on line 395 from 8Gi to 50Gi
3) `helm upgrade mongo bitnami/mongodb -f values.yml`
See error:
```
Error: UPGRADE FAILED: cannot patch "mongodb" with kind StatefulSet: StatefulSet.apps "mongodb" is invalid: spec: Forbidden: updates to statefulset spec for fields other than 'replicas', 'template', and 'updateStrategy' are forbidden
```

### What is the feature you are proposing to solve the problem?

I expect that it should upgrade expand the volume

### What alternatives have you considered?

Possible instruction or workarounds

**Comment by joancafom** [comment]
Hi @igorrudyk 

I believe this is not directly related to the chart itself and has to do more with Kubernetes instead. You can check more info in the official documentation:

https://kubernetes.io/docs/concepts/storage/persistent-volumes/#resizing-a-volume-containing-a-file-system
https://kubernetes.io/docs/concepts/storage/persistent-volumes/#resizing-an-in-use-persistentvolumeclaim

**Comment by igorrudyk** [comment]
Thank you, @joancafom

**Comment by rudxde** [comment]
This article lead me to an working process for updating the stateful set volume size:
https://www.giffgaff.io/tech/resizing-statefulset-persistent-volumes-with-zero-downtime
https://kubernetes.io/docs/concepts/storage/persistent-volumes/#resizing-a-volume-containing-a-file-system

TLDR:
* ensure the stoage class has set `allowVolumeExpansion` to true
* Delete statefulset without deleting its pods `kubectl delete sts <statefulset-name> --cascade=orphan`
* Modify the Persistentvolumeclaims
* recreate statefulset with new sorage size (helm deploy etc)
* delete each pod after another
